java-persistence/src/main/java/com/whitecloak/training/inventory/gateway/impl/ItemGatewayImpl.java

```diff
@@ -1,34 +1,104 @@
+import com.whitecloak.training.inventory.model.User;
 import com.whitecloak.training.inventory.persistence.entity.ItemEntity;
+import com.whitecloak.training.inventory.persistence.repository.CategoryRepository;
 import com.whitecloak.training.inventory.persistence.repository.ItemRepository;
+import com.whitecloak.training.inventory.persistence.repository.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
 
 @Component
 class ItemGatewayImpl implements ItemGateway {
 
     private final ItemRepository itemRepository;
+    private final CategoryRepository categoryRepository;
+    private final UserRepository userRepository;
 
     @Autowired
-    ItemGatewayImpl(ItemRepository itemRepository) {
+    ItemGatewayImpl(ItemRepository itemRepository,
+                    CategoryRepository categoryRepository,
+                    UserRepository userRepository) {
         this.itemRepository = itemRepository;
+        this.categoryRepository = categoryRepository;
+        this.userRepository = userRepository;
     }
 
     @Override
-    public Item create(String name) {
+    public Item create(ItemForm form) {
+
         ItemEntity entity = new ItemEntity();
-        entity.setOwnerId(99L);
-        entity.setName(name);
+        entity.setName(form.getName());
+        entity.setCategory(categoryRepository.findById(form.getCategoryId()));
+        entity.setOwner(userRepository.findById(form.getOwnerId()));
 
         ItemEntity savedEntity = itemRepository.save(entity);
+        Item item = mapToModel(savedEntity);
+
+        return item;
+    }
+
+    @Override
+    public Page<Item> all(int page, int size) {
+        Pageable pageable = PageRequest.of(page, size);
+        Page<ItemEntity> entities =  itemRepository.findAllByDeleted(false, pageable);
+        Page<Item> items = entities.map((e) -> mapToModel(e));
+        return items;
+    }
+
+    @Override
+    public Page<Item> all(String category, int page, int size) {
+        Pageable pageable = PageRequest.of(page, size);
+        Page<ItemEntity> entities =  itemRepository.findAllByDeletedAndCategory_Name(false, category, pageable);
+        Page<Item> items = entities.map((e) -> mapToModel(e));
+        return items;
+    }
+
+    @Override
+    public Item get(Long id) {
+        ItemEntity entity = itemRepository.findByDeletedAndId(false, id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
 
+        Item item = mapToModel(entity);
+        return item;
+    }
+
+    @Override
+    public void delete(Long id) {
+        ItemEntity entity = itemRepository.findByDeletedAndId(false, id).map((e) -> {
```

Comment:
You are misusing the `map()` operation.
Mapping is for transforming an object to another object, not setting an object's state.

It would be proper to:

```java
ItemEntity entity = itemRepository.findByDeletedAndId(false, id)
    .orElseThrow( /* ... */ );
entity.setDeleted(true);
itemRepository.save(entity);
```